gusto/time_discretisation/time_discretisation.py

```diff
@@ -374,6 +382,13 @@ def setup(self, equation, apply_bcs=True, *active_labels):
             self.ncycles = 1
         self.x0 = Function(self.fs)
         self.x1 = Function(self.fs)
+        # If the time_derivative term is nonlinear, we must use a nonlinear solver
+        if (len(self.residual.label_map(lambda t: t.has_label(nonlinear_time_derivative), map_if_false=drop)) > 0
```

Comment:
Could we separate this over multiple lines?

Reply:
Done!